tests/integration/targets/aci_l3out/tasks/main.yml

```diff
@@ -177,19 +201,26 @@
         - nm_add_l3out_ospf.sent.l3extOut.children.2.ospfExtP.attributes.areaType == "regular"
         - nm_add_l3out_ospf.sent.l3extOut.children.2.ospfExtP.attributes.descr == "test for ospf protocol"
         - nm_add_l3out_ospf.sent.l3extOut.children.2.ospfExtP.attributes.multipodInternal == "no"
+        - nm_add_l3out_wout_ospf_spec.msg == "ospf specifications are required when l3protocol is 'ospf'"
 
   - name: get all in l3out
     aci_l3out:
       <<: *aci_l3out_present
       state: query
+    register: get_l3out
+
+  - name: get all l3out
+    aci_l3out:
+      <<: *aci_info
+      state: query
     register: get_all_l3out
 
   - name: asserts query tasks
     assert:
       that:
-        - get_all_l3out is not changed
-        - '"rsp-subtree=full&rsp-subtree-class=bgpExtP,eigrpExtP,l3extRsEctx,l3extRsL3DomAtt,ospfExtP,pimExtP" in get_all_l3out.filter_string'
-        - '"tn-ansible_tenant/out-ansible_l3out.json" in get_all_l3out.url'
+        - get_l3out is not changed
```

Comment:
When retrieving multiple objects you should assert also that all these expected l3out are returned. Same applies for one return. See similar example here: https://github.com/CiscoDevNet/ansible-aci/blob/master/tests/integration/targets/aci_access_span_src_group/tasks/main.yml#L249